plan/logical_plan_builder.go

```diff
@@ -1141,6 +1149,42 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 			RetType:  &col.FieldType,
 			Position: i,
 			ID:       col.ID})
+		if tableInfo.PKIsHandle && mysql.HasPriKeyFlag(col.Flag) {
+			pkCol = schema.Columns[schema.Len()-1]
+		}
+	}
+	needUnionScan := b.ctx.Txn() != nil && !b.ctx.Txn().IsReadOnly()
+	if b.needColHandle || needUnionScan {
+		if pkCol == nil || needUnionScan {
+			idCol := &expression.Column{
+				FromID:   p.id,
+				DBName:   schemaName,
+				TblName:  tableInfo.Name,
+				ColName:  model.NewCIStr("_rowid"),
+				RetType:  types.NewFieldType(mysql.TypeLonglong),
+				Position: schema.Len(),
+				Index:    schema.Len(),
+				ID:       -1,
```

Comment:
The column's ID is `-1` means that the column is a handle column?
Can we define a constant instead of `-1`?